File: Ruby/chap01/01-age.rb
require  'Qt'

app = Qt::Application.new(ARGV)

window = Qt::Widget.new
window.setWindowTitle("Enter Your Age")

spinbox = Qt::SpinBox.new
slider  = Qt::Slider.new(Qt::Horizontal)
spinbox.setRange(0, 130)
spinbox.setValue(35)
slider.setRange(0, 130)
Qt::Object.connect(spinbox, SIGNAL('valueChanged(int)'),
                   slider, SLOT('setValue(int)') )
Qt::Object.connect(slider, SIGNAL('valueChanged(int)'),
                   spinbox, SLOT('setValue(int)'))

layout = Qt::HBoxLayout.new
layout.addWidget(spinbox)
layout.addWidget(slider)

window.setLayout(layout)
window.show

app.exec

File: Python/chap01/01-Age.py
import sys
from PyQt4.QtCore import *
from PyQt4.QtGui import *

app = QApplication(sys.argv)

window = QWidget()
window.setWindowTitle("Enter Your Age")

spinbox = QSpinBox()
slider = QSlider(Qt.Horizontal)
spinbox.setRange(0, 130)
slider.setRange(0,130)
spinbox.valueChanged.connect(slider.setValue)
slider.valueChanged.connect(spinbox.setValue)
spinbox.setValue(35)

layout = QHBoxLayout()
layout.addWidget(spinbox)
layout.addWidget(slider)

window.setLayout(layout)
window.show()

app.exec_()